dvc/objects/db/base.py

```diff
@@ -140,9 +167,16 @@ def check(self, hash_info, check_hash=True):
         - Remove the file from cache if it doesn't match the actual hash
         """
 
+        if check_staging and self.staging is not None:
+            try:
+                self.staging.check(hash_info, check_hash=check_hash)
+                return
+            except Exception:  # pylint: disable=broad-except
```

Comment:
Could you elaborate why it needs to be so broad, please?

Reply:
If we can't find the hash for any reason in the staging area, we fall back to checking the parent ODB